### PR TITLE
Migrated from package_info to package_info_plus community package

### PR DESCRIPTION
**Pull request by JigneshWorld**
## :scroll: Description

Migrated to package_info_plus plugin

## :bulb: Motivation and Context

We can use this PR and plugin code as a base to add support for other desktop platform with contextual info

## :green_heart: How did you test it?
Executed `./tool/presubmit.sh` within the `dart` and `flutter` folders

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes - AFAIK: not needed any additional tests
- [ ] I updated the docs if needed - AFAIK: not needed any updates on docs
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 * Fix: Enable breadcrumb recording mechanism based on platform (#366)
 * Feat: Send default PII options (#360)
 * Bump: sentry-cocoa to v6.2.1 (#360)
+* Feat: Migration from `package_info` to `package_info_plus` plugin
```
Comment (marandaneto) [inline CHANGELOG.md]:
```suggestion
* Feat: Migration from `package_info` to `package_info_plus` plugin (#370)
```

Comment (JigneshWorld) [reply CHANGELOG.md]: Thanks for your quick suggestion @marandaneto 👍

**Comment by ueman** [comment]
If the Flutter team itself is advocating to use this package, I see no problem with this change. 
[package_info_plus](https://pub.dev/packages/package_info_plus) also supports more platforms than [package_info](https://pub.dev/packages/package_info). 

Does package_info_plus have higher minimum OS requirements? That could be one problem, but I haven't looked that up yet.

**Comment by JigneshWorld** [comment]
> If the Flutter team itself is advocating to use this package, I see no problem with this change.
> [package_info_plus](https://pub.dev/packages/package_info_plus) also supports more platforms than [package_info](https://pub.dev/packages/package_info).
> 
> Does package_info_plus have higher minimum OS requirements? That could be one problem, but I haven't looked that up yet.

I have just checked to confirm both has same minimum OS requirements.
> `ios -> 8.0` 
> `android:minSdkVersion -> 16`

**Comment by marandaneto** [comment]
its fine then because Flutter is min 16 itself.
I'll review & test a bit next week, a bit tight on this one, thanks a lot @JigneshPatel23

**Review by marandaneto — APPROVED** [review]
LGTM, thanks @JigneshPatel23 :)

**Comment by marandaneto** [comment]
@JigneshPatel23 CI is failing for some reason, not sure if its related to the change though

**Comment by JigneshWorld** [comment]
Yes, just checked the details for failed, seems like there is nothing related to the changes.
@marandaneto 
Just for reference, other open PR also failing for the same reason on CI, something needs to be looked in general(CI or build).

**Comment by ueman** [comment]
> Yes, just checked the details for failed, seems like there is nothing related to the changes.
> @marandaneto
> Just for reference, other open PR also failing for the same reason on CI, something needs to be looked in general(CI or build).

Yep, I created https://github.com/getsentry/sentry-dart/issues/377 for that.

**Review by bruno-garcia — APPROVED** [review]

**Comment by marandaneto** [comment]
@JigneshPatel23 mind pulling `main` to your branch? should fix it, thanks

**Comment by JigneshWorld** [comment]
@marandaneto sure, CI is now happy 👍

**Comment by paulononaka** [comment]
Hi. When will it be published on pub.dev? Thanks.

**Comment by marandaneto** [comment]
> Hi. When will it be published on pub.dev? Thanks.

https://github.com/getsentry/sentry-dart/releases/tag/5.0.0

**Pull request merged**